tests/ping.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
+	"k8s.io/utils/net"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+var (
+	promptPriv   = regexp.MustCompile("\\# ")
+	promptUnPriv = regexp.MustCompile("\\$ ")
+	shellSuccess = regexp.MustCompile("\n0")
+	shellFail    = regexp.MustCompile("\n[1-9].*")
+)
+
+// PingFromVMConsole performs a ping through the provided VMI console.
+// Optional arguments for the ping command may be provided, overwirtting the defaul ones.
+// (default ping options: "-c 1, -w 5")
+// Note: The maximum overall command timeout is 10 seconds.
+func PingFromVMConsole(vmi *v1.VirtualMachineInstance, ipAddr string, args ...string) error {
+	pingString := "ping"
+	if net.IsIPv6String(ipAddr) {
+		pingString = "ping -6"
+	}
+
+	if len(args) == 0 {
+		args = []string{"-c 1", "-w 5"}
+	}
+	args = append([]string{pingString, ipAddr}, args...)
+	cmdCheck := strings.Join(args, " ") + "\n"
+
+	err := CheckForTextExpecter(vmi, []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		expectPrompt(),
+		&expect.BSnd{S: cmdCheck},
+		expectPrompt(),
+		&expect.BSnd{S: "echo $?\n"},
+		&expect.BCas{C: []expect.Caser{
+			&expect.Case{
+				R: shellSuccess,
+				T: expect.OK(),
+			},
+			&expect.Case{
+				R: shellFail,
+				T: expect.Fail(expect.NewStatus(codes.Unavailable, "ping failed")),
+			},
+		}},
+	}, 10)
+	if err != nil {
+		return fmt.Errorf("Failed to ping VMI %s, error: %v", vmi.Name, err)
+	}
+	return nil
+}
+
+func expectPrompt() *expect.BCas {
```

Comment:
I suppose this does an "or" of possible prompts but are we sure that the test don't want to really check one or the other ? At the end one option we can pass to the function is expect.BCas and by default is this.

Reply:
ping does not require a specific prompt, it can run from any so I see no need to do this.

Reply:
Ack